java/src/jmri/web/servlet/operations/HtmlManifest.java

```diff
@@ -228,7 +230,17 @@ public String getLocations() throws IOException {
     protected String blockCars(JsonNode cars, RouteLocation location, boolean isManifest) {
         StringBuilder builder = new StringBuilder();
         log.debug("Cars is {}", cars);
-        for (JsonNode car : cars.path(JSON.ADD)) {
+
+        //copy the adds into a sortable arraylist
+        ArrayList<JsonNode> adds = new ArrayList<JsonNode>();
+        cars.path(JSON.ADD).forEach(adds::add);
+            
+        //sort if requested
+        if (adds.size() > 0 && Setup.isSortByTrackNameEnabled()) {
+            adds.sort(Comparator.comparing(o -> o.get("location").get("track").get("userName").asText()));
```

Comment:
Just checking: Is `get("userName")` always returning a string and never null?

Reply:
I'm not sure. I'll investigate. I'll switch to use .path() instead of .get() as a safer alternative.